safe/utilities/analysis.py

```diff
@@ -125,12 +125,16 @@ def user_extent(self):
     @user_extent.setter
     def user_extent(self, user_extent):
         # We transfer the extent to the IF as a list.
-        extent = [
-            user_extent.xMinimum(),
-            user_extent.yMinimum(),
-            user_extent.xMaximum(),
-            user_extent.yMaximum()]
-        self.impact_function.requested_extent = extent
+        if isinstance(user_extent, QgsRectangle):
+            extent = [
+                user_extent.xMinimum(),
+                user_extent.yMinimum(),
+                user_extent.xMaximum(),
+                user_extent.yMaximum()]
+            self.impact_function.requested_extent = extent
+        else:
```

Comment:
Will this ever happen?
And if so, is the exception handled properly, and beautifully?